cmd/kured/main.go

```diff
@@ -166,12 +165,20 @@ func main() {
 func flagCheck(cmd *cobra.Command, args []string) {
 	if slackHookURL != "" && notifyURL != "" {
 		log.Warnf("Cannot use both --notify-url and --slack-hook-url flags. Kured will use --notify-url flag only...")
-		slackHookURL = ""
 	}
-	if slackChannel != "" || slackHookURL != "" || slackUsername != "" {
+
+	if slackHookURL != "" {
 		log.Warnf("Deprecated flag(s). Please use --notify-url flag instead.")
+		tmpsplit := strings.Split(slackHookURL, "/")
+		if len(tmpsplit) < 4 {
+			log.Warnf("slack-hook-url is not properly formatted...\nno notification will be sent.")
+		} else {
+			tokenA := tmpsplit[len(tmpsplit)-3]
+			tokenB := tmpsplit[len(tmpsplit)-2]
+			tokenC := tmpsplit[len(tmpsplit)-1]
```

Comment:
Would it make sense to add a couple of obvious examples in a unit test to check that this works?

Reply:
yes, I would extract a bit of that logic into a unit testable function, and add the tests. Agree with @dholbach . Or just unit test the flagCheck into one go, which sounds better... (However, we only have one series of cli flags checked in that form for now! We should probably check for the existence of the command for reboot or for sentinel, but this is unrelated to this PR...)

Next to that, I would add a comment before the L166 and L170 to tell that this code should be removed in the next release: We probably want to clean up this "automatic migration" in the future.